studyBuddy/studyBuddy/dataNeeds/UserDataFetcher.cs

```diff
@@ -11,44 +11,79 @@ class UserDataFetcher : UserDataAbstract
 
         public string GetSalt(string username)
         {
-            string[] row = source.SelectOneRow("salt,id FROM " + MysqlHandler.tblUsers +
+            using (var db = new studybuddyDBEntities() )
+            {
+                var user = db.Users.Where(use => use.username.ToLower() == username.ToLower()).DefaultIfEmpty(null).FirstOrDefault();
+                
+                userId = user.ID;
+                return user.salt;
+
+            }
+            /*string[] row = source.SelectOneRow("salt,id FROM " + MysqlHandler.tblUsers +
                 $" WHERE username = '{username}' ;");
             if (row.Length == 0)
                 return "";
             userId = Convert.ToInt32(row[1]);
-            return row[0];
+            return row[0];*/
         }
 
         public string GetSalt(System.Net.Mail.MailAddress mail)
         {
-            string[] row = source.SelectOneRow("salt,id FROM " + MysqlHandler.tblUsers +
-                $" WHERE email = '{mail.Address}' ;");
-            if (row.Length == 0)
-                return "";
-            userId = Convert.ToInt32(row[1]);
-            return row[0];
+            using(var db = new studybuddyDBEntities())
+            {
+                var user = db.Users.Where(use => use.email.ToLower() == mail.Address.ToLower()).DefaultIfEmpty(null).FirstOrDefault();
+                
```

Comment:
Should check at first if(user == null)

Reply:
Oh my. I will correct all simillar cases